autode/bond_rearrangement.py

```diff
@@ -10,6 +13,211 @@
     find_cycles,
 )
 
+if TYPE_CHECKING:
+    from autode.species.species import Species
+    import networkx as nx
+
+
+_MAX_BOND_REARR = 4  # maximum allowed bond rearrangment
+
+
+class BondRearrGenerator:
+    def __init__(
+        self,
+        reactant: "Species",
+        product: "Species",
+        delta_bond_tot: int,
+        extra_move_pairs: int = 0,
+    ):
+        """
+        Create a class that obtains all possible bond rearrangements that
+        will transform the reactant into the product based on the types of
+        bonds that must be broken and formed
+
+        Args:
+            reactant:
+            product:
+            delta_bond_tot: Total number of bonds changing from reactant to product
+            extra_move_pairs: Number of pairs of extra graph moves that maybe made
+                              in addition to the known types
+        """
+        self._rct_bond_dict: Optional[dict[str, list]] = None
+
+        self._reactant = reactant
+        self._product = product
+        assert isinstance(delta_bond_tot, int)
+
+        # number of extra moves must be even i.e. in pairs (form + break)
+        assert extra_move_pairs >= 0
```

Comment:
if the comment is accurate
```suggestion
        assert extra_move_pairs >= 0 and extra_move_pairs % 2 == 0
```

Reply:
@t-young31 Sorry, the comment was probably unclear, it should be the number of pairs. So if it is 1, that means two extra moves are made.